example/gpu_memory_profiler/README.md

```diff
@@ -0,0 +1,146 @@
+
+Machine learning training tasks running on the GPUs are very frequently limited 
+  by the GPU memory capacity, frontend programmers therefore need a GPU memory 
+  profiler to understand where the memory goes.
+The problems with the existing GPU memory profilers are that they are too high-level 
+  (similar to `nvidia-smi` that only provides a sum of all the consumptions) 
+  and hence make it challenging for the frontend programmers to get the big picture.
+
+In this example, we provide instructions on how to use the **MXNet GPU memory profiler**.
+We start by making the observation that most MXNet programmers have
+  a very good habit of assigning names to computation symbols:
+
+```Python
+# Symbolic Graph Implementation of an LSTM Cell
+i2h = symbol.FullyConnected(data=inputs, weight=self._iW, bias=self._iB,
+                            num_hidden=self._num_hidden*4,
+                            name='%si2h'%name)
+h2h = symbol.FullyConnected(data=states[0], weight=self._hW, bias=self._hB,
+                            num_hidden=self._num_hidden*4,
+                            name='%sh2h'%name)
+```
+
+Those names contain rich information regarding the model that is being described 
+  and should be leveraged during the GPU memory profiling phase.
+Luckily, MXNet already has the ability of propagating this information 
+  all the way down from the Python APIs to its C++ core `nnvm::Graph`.
+When the graph executor tries to initialize its data entries, we extract such information 
+  out from the computation graph to tag the data entries (of type `mxnet::NDArray`),
+  and when those data entries are materialized we are able to propagate those names
+  to the storage allocators and record them inside the logging files,
+  which could be further used for visualization and/or analyzing purposes.
+
+![MXNet-GPU_Memory_Profiler-Design](https://github.com/dmlc/web-data/blob/master/mxnet/example/gpu_memory_profiler/MXNet-GPU_Memory_Profiler-Design.png?raw=true)
+
+
+## Instructions
+
+*The video below was demoed at **SysML 2019** ([Demo #24](https://www.sysml.cc/doc/2019/demo_24.pdf)). It shows the expected behavior of the GPU memory profiler. Please kindly note that we have made several changes since the demo so the current version looks slightly different.*
+
+![MXNet-GPU_Memory_Profiler](https://github.com/dmlc/web-data/blob/master/mxnet/example/gpu_memory_profiler/MXNet-GPU_Memory_Profiler.gif?raw=true)
+
+*In order for the GPU memory profiler to be enabled, you need to **compile from source**.*
+
+- Download the MXNet codebase and install the prerequisite software libraries, as is instructed
+    [here](https://mxnet.incubator.apache.org/versions/master/install/build_from_source.html).
+
+- Modify the `MXNET_ENABLE_STORAGE_TAGGING` flag in `include/storage_tag.h` to **1**,
+    which controls the storage tagging (disabled by default),
+    **and** `MXNET_USE_GPU_MEMORY_PROFILER` flag in `src/profiler/gpu_memory_profiler.h` **1**,
```

Comment:
I tried this. I was able to generate csv and log. But I wasnt able to use the analyzer. I think I am using SETME wrongly. Some examples in SETME would help.

Reply:
I agree that adding a microbenchmark might be helpful in this case. Added to the TODO list.